src/org/thoughtcrime/securesms/database/LokiBackupFilesDatabase.kt

```diff
@@ -0,0 +1,119 @@
+package org.thoughtcrime.securesms.database
+
+import android.content.ContentValues
+import android.content.Context
+import android.database.Cursor
+import android.net.Uri
+import androidx.annotation.NonNull
+import org.thoughtcrime.securesms.database.helpers.SQLCipherOpenHelper
+import org.thoughtcrime.securesms.database.model.BackupFileRecord
+import java.lang.IllegalArgumentException
+import java.util.*
+import kotlin.collections.ArrayList
+
+/**
+ * Keeps track of the backup files saved by the app.
+ * Uses [BackupFileRecord] as an entry data projection.
+ */
+class LokiBackupFilesDatabase(context: Context?, databaseHelper: SQLCipherOpenHelper?)
```

Comment:
The `context` parameter here isn't \* really \* optional. The compiler will tell you that the argument for the superclass is optional because it's coming from Java, but if you pass in `null` it won't work.

Reply:
Same for `databaseHelper`